### PR TITLE
RTL support for Calc

### PR DESCRIPTION
**Pull request by dennisfrancis**
* Target version: master 

### Summary
RTL support for Calc

This needs core side patches from https://cgit.freedesktop.org/libreoffice/core/log/?h=feature/lok-calc-rtl

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

**Review by gokaysatir — APPROVED** [review]
Thanks for the great improvements:)

File: browser/src/control/Control.AutofilterDropdown.js
Comment (Ashod) [inline browser/src/control/Control.AutofilterDropdown.js:135]: Shouldn't we always have braces?

File: browser/src/control/Control.ColumnHeader.js
Comment (Ashod) [inline browser/src/control/Control.ColumnHeader.js:217]: Same. I think having braces around even single-line scopes is the convention used in .js/.ts.

File: browser/src/layer/tile/TilesSection.ts
Comment (Ashod) [inline browser/src/layer/tile/TilesSection.ts:650]: It would seem toScaleAbs is not used. Is this a bug?

File: browser/src/layer/vector/CanvasOverlay.ts
Comment (Ashod) [inline browser/src/layer/vector/CanvasOverlay.ts:108]: The single `s` at the end is a very small difference. This wouldn't be a problem if there were no `transform` (without `s`) elsewhere. But since we are using both variables with a single character difference, sooner or later someone is going to have a typo and cause a bug.

Perhaps we should make this `trasnformList` to avoid confusion and accidental bugs ?

File: browser/src/layer/vector/SVGGroup.js
Comment (Ashod) [inline browser/src/layer/vector/SVGGroup.js:79]: This exists in other file(s) too. Perhaps it's better to make this part of `docLayer` and replace it with a single `this._map._docLayer.isCalcRTL()` call? Sure it's not great to have application-specific helpers in docLayer (calc, in this case), but copy pasting these 4 lines is probably worse.

File: browser/src/layer/vector/SVGGroup.js
Comment (Ashod) [inline browser/src/layer/vector/SVGGroup.js:144]: Braces?

File: browser/src/layer/vector/SplitPanesSVG.js
Comment (Ashod) [inline browser/src/layer/vector/SplitPanesSVG.js:77]: Consider making this part of `docLayer`. Too many copy/pastes.

File: browser/src/map/Map.js
Comment (Ashod) [inline browser/src/map/Map.js:992]: This describes what the function does, which is easy to read/see, but it doesn't explain why or when/where this is useful. It would be helpful to maintainers to read a note as to when this is typically used so they can more easily understand the purpose of this function. I.e. the 'why' as opposed to the 'what'.

File: kit/KitHelper.hpp
Comment (Ashod) [inline kit/KitHelper.hpp:96]: Of course we could have a single-char instead of a literal-string here:
```
rtlposs << i << ',';
```

**Review by Ashod — COMMENTED** [review]
Thanks @dennisfrancis! Left some notes. Hope they are helpful.

File: browser/src/control/Control.AutofilterDropdown.js
Comment (mmeeks) [reply browser/src/control/Control.AutofilterDropdown.js]: Ideally not - so the code is readable ;-)

**Comment by timar** [comment]
@gokaysatir gave +1 before my patch with the braces, so we can merge. 
@dennisfrancis could you please read @Ashod 's notes, and if you agree, make those changes in a follow-up commit.

File: browser/src/layer/tile/TilesSection.ts
Comment (dennisfrancis) [reply browser/src/layer/tile/TilesSection.ts:651]: These lines are not from this PR, but from older zoom-animation related changes. Looks like they weren't removed after factoring some code to private utility functions (scalePosForZoom etc). I'll remove them in a follow up.

File: browser/src/layer/vector/CanvasOverlay.ts
Comment (dennisfrancis) [reply browser/src/layer/vector/CanvasOverlay.ts:108]: This particular variable is a private member of the CanvasOverlay class. The other transform variable is a local variable of a different type (OverlayTransform). The typescript transpiler will complain if anyone uses `transforms` instead of `transform` or vice-versa. But I agree that renaming `transforms` to `transformList` makes it more distinctive and readable.

**Comment by dennisfrancis** [comment]
https://github.com/CollaboraOnline/online/pull/3972 contains the follow-up patches.

**Pull request merged**
